src/opm/input/eclipse/Schedule/WellTraj/RigEclipseWellLogExtractor.cpp

```diff
@@ -0,0 +1,325 @@
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleGrid.hpp>
+
+#include <opm/input/eclipse/Schedule/WellTraj/RigEclipseWellLogExtractor.hpp>
+#include <external/resinsight/ReservoirDataModel/RigWellLogExtractionTools.h>
+#include <external/resinsight/ReservoirDataModel/RigWellPath.h>
+#include <external/resinsight/ReservoirDataModel/cvfGeometryTools.h>
+#include <external/resinsight/ReservoirDataModel/RigWellLogExtractor.h>
+#include <external/resinsight/ReservoirDataModel/RigCellGeometryTools.h>
+#include <external/resinsight/CommonCode/cvfStructGrid.h>
+#include <external/resinsight/LibGeometry/cvfBoundingBox.h>
+
+#include <map>
+
+namespace external {
+//==================================================================================================
+///
+//==================================================================================================
+
+ RigEclipseWellLogExtractor::RigEclipseWellLogExtractor(const RigWellPath* wellpath, const Opm::EclipseGrid& grid, cvf::ref<cvf::BoundingBoxTree>& cellSearchTree)
+    : RigWellLogExtractor( wellpath, "" )
+      ,m_grid(grid)
+      ,m_cellSearchTree(cellSearchTree)
+{
+    calculateIntersection();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RigEclipseWellLogExtractor::calculateIntersection()
+{
+    std::map<RigMDCellIdxEnterLeaveKey, HexIntersectionInfo> uniqueIntersections;
+
+    bool isCellFaceNormalsOut = 1; //m_caseData->mainGrid()->isFaceNormalsOutwards();
+
+    if ( m_wellPathGeometry->wellPathPoints().empty() ) return;
+
+    for ( size_t wpp = 0; wpp < m_wellPathGeometry->wellPathPoints().size() - 1; ++wpp )
+    {
+        std::vector<HexIntersectionInfo> intersections;
+        cvf::Vec3d                       p1 = m_wellPathGeometry->wellPathPoints()[wpp];
+        cvf::Vec3d                       p2 = m_wellPathGeometry->wellPathPoints()[wpp + 1];
+
+        cvf::BoundingBox bb;
+
+        bb.add( p1 );
+        bb.add( p2 );
+
+        buildCellSearchTree();
+        
+        std::vector<size_t> closeCellIndices = findCloseCellIndices( bb );
+
+        cvf::Vec3d hexCorners_opm[8]; //opm numbering
+        std::array<double, 3> cornerPointArray;
```

Comment:
Maybe make this a local variable of the innermost loop?

Reply:
Done